src/FormTaxonomies/resources/form-builder/form-categories.tsx

```diff
@@ -0,0 +1,54 @@
+import {getAvailableFormCategories, getInitialFormCategories} from "./windowData";
+import {buildTermsTree} from "./utils/terms";
+import {CheckboxControl} from "@wordpress/components";
+import {decodeEntities} from "@wordpress/html-entities";
+
+
+const FormCategorySetting = ({settings, setSettings}) => {
+    const {
+        formCategories = getInitialFormCategories(),
+    } = settings;
+
+    const categoryTree = buildTermsTree(getAvailableFormCategories());
+
+    const onChange = (categoryId ) => {
+        setSettings({formCategories: formCategories.includes( categoryId )
+                ? formCategories.filter( ( id ) => id !== categoryId )
+                : [ ...formCategories, categoryId ]
+        })
+    };
+
+    const renderTerms = (renderedTerms) => {
+        return renderedTerms.map((term) => {
```

Comment:
Creating functions inside a react component means it will be re-created every re-render.  Some ways to avoid this would be to use [useCallback](https://react.dev/reference/react/useCallback), move the entire function outside of the component, or don't create a function and just return the map.

Reply:
I was able to refactor the function so that it could be extracted (and is no longer coupled with local variables in the component context, see [53f4b0d](https://github.com/impress-org/givewp/pull/7463/commits/53f4b0d08d30f03d587b6922e22a7965b2c50f38).

Reply:
Nice, thanks!